src/common/utxobased/keymanager/bip69.ts

```diff
@@ -0,0 +1,31 @@
+/***
+ * https://github.com/bitcoin/bips/blob/master/bip-0069.mediawiki
+ */
+
+import { Input, Output } from './utxopicker'
+
+/***
+ * Previous transaction hashes (in reversed byte-order) are to be sorted in ascending order, lexicographically.
+ * In the event of two matching transaction hashes,
+ * the respective previous output indices will be compared by their integer value, in ascending order.
+ * If the previous output indices match, the inputs are considered equal.
+ */
+export const sortInputs = (inputs: Input[]): Input[] =>
+  [...inputs].sort((a, b) => {
+    const aHash = Buffer.from(a.hash).reverse()
```

Comment:
Remove the `reverse` method invocation:

```js
const sortInputs = (inputs) =>
[...inputs].sort((a, b) => {
  const aHash = Buffer.from(a.hash)
  const bHash = Buffer.from(b.hash)
  const compare = aHash.compare(bHash)
  return compare !== 0 ? compare : a.index - b.index
})

sortInputs([
  { hash: "0e53ec5dfb2cb8a71fec32dc9a634a35b7e24799295ddd5278217822e0b31f57", index: 0 },
  { hash: "26aa6e6d8b9e49bb0630aac301db6757c02e3619feb4ee0eea81eb1672947024", index: 1 },
  { hash: "28e0fdd185542f2c6ea19030b0796051e7772b6026dd5ddccd7a2f93b73e6fc2", index: 0 },
  { hash: "381de9b9ae1a94d9c17f6a08ef9d341a5ce29e2e60c36a52d333ff6203e58d5d", index: 1 },
  { hash: "3b8b2f8efceb60ba78ca8bba206a137f14cb5ea4035e761ee204302d46b98de2", index: 0 },
  { hash: "402b2c02411720bf409eff60d05adad684f135838962823f3614cc657dd7bc0a", index: 1 },
  { hash: "54ffff182965ed0957dba1239c27164ace5a73c9b62a660c74b7b7f15ff61e7a", index: 1 },
  { hash: "643e5f4e66373a57251fb173151e838ccd27d279aca882997e005016bb53d5aa", index: 0 },
  { hash: "6c1d56f31b2de4bfc6aaea28396b333102b1f600da9c6d6149e96ca43f1102b1", index: 1 },
  { hash: "7a1de137cbafb5c70405455c49c5104ca3057a1f1243e6563bb9245c9c88c191", index: 0 },
  { hash: "7d037ceb2ee0dc03e82f17be7935d238b35d1deabf953a892a4507bfbeeb3ba4", index: 1 },
  { hash: "a5e899dddb28776ea9ddac0a502316d53a4a3fca607c72f66c470e0412e34086", index: 0 },
  { hash: "b4112b8f900a7ca0c8b0e7c4dfad35c6be5f6be46b3458974988e1cdb2fa61b8", index: 0 },
  { hash: "bafd65e3c7f3f9fdfdc1ddb026131b278c3be1af90a4a6ffa78c4658f9ec0c85", index: 0 },
  { hash: "de0411a1e97484a2804ff1dbde260ac19de841bebad1880c782941aca883b4e9", index: 1 },
  { hash: "f0a130a84912d03c1d284974f563c5949ac13f8342b8112edff52971599e6a45", index: 0 },
  { hash: "f320832a9d2e2452af63154bc687493484a0e7745ebd3aaf9ca19eb80834ad60", index: 0 },
])
```

Results:

```js
[
  {
    hash: '0e53ec5dfb2cb8a71fec32dc9a634a35b7e24799295ddd5278217822e0b31f57',
    index: 0
  },
  {
    hash: '26aa6e6d8b9e49bb0630aac301db6757c02e3619feb4ee0eea81eb1672947024',
    index: 1
  },
  {
    hash: '28e0fdd185542f2c6ea19030b0796051e7772b6026dd5ddccd7a2f93b73e6fc2',
    index: 0
  },
  {
    hash: '381de9b9ae1a94d9c17f6a08ef9d341a5ce29e2e60c36a52d333ff6203e58d5d',
    index: 1
  },
  {
    hash: '3b8b2f8efceb60ba78ca8bba206a137f14cb5ea4035e761ee204302d46b98de2',
    index: 0
  },
  {
    hash: '402b2c02411720bf409eff60d05adad684f135838962823f3614cc657dd7bc0a',
    index: 1
  },
  {
    hash: '54ffff182965ed0957dba1239c27164ace5a73c9b62a660c74b7b7f15ff61e7a',
    index: 1
  },
  {
    hash: '643e5f4e66373a57251fb173151e838ccd27d279aca882997e005016bb53d5aa',
    index: 0
  },
  {
    hash: '6c1d56f31b2de4bfc6aaea28396b333102b1f600da9c6d6149e96ca43f1102b1',
    index: 1
  },
  {
    hash: '7a1de137cbafb5c70405455c49c5104ca3057a1f1243e6563bb9245c9c88c191',
    index: 0
  },
  {
    hash: '7d037ceb2ee0dc03e82f17be7935d238b35d1deabf953a892a4507bfbeeb3ba4',
    index: 1
  },
  {
    hash: 'a5e899dddb28776ea9ddac0a502316d53a4a3fca607c72f66c470e0412e34086',
    index: 0
  },
  {
    hash: 'b4112b8f900a7ca0c8b0e7c4dfad35c6be5f6be46b3458974988e1cdb2fa61b8',
    index: 0
  },
  {
    hash: 'bafd65e3c7f3f9fdfdc1ddb026131b278c3be1af90a4a6ffa78c4658f9ec0c85',
    index: 0
  },
  {
    hash: 'de0411a1e97484a2804ff1dbde260ac19de841bebad1880c782941aca883b4e9',
    index: 1
  },
  {
    hash: 'f0a130a84912d03c1d284974f563c5949ac13f8342b8112edff52971599e6a45',
    index: 0
  },
  {
    hash: 'f320832a9d2e2452af63154bc687493484a0e7745ebd3aaf9ca19eb80834ad60',
    index: 0
  }
]
```

Reply:
Discovery: The hash Buffer is little-endian so must be reversed in order to turn it into the txid (big-endian). The implementation is correct.